File: go-client-sdk/pkg/models/operations/getcompaniescompanyidreportsageddebtor.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDReportsAgedDebtorPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDReportsAgedDebtorQueryParams struct {
	NumberOfPeriods  *int       `queryParam:"style=form,explode=true,name=numberOfPeriods"`
	PeriodLengthDays *int       `queryParam:"style=form,explode=true,name=periodLengthDays"`
	ReportDate       *time.Time `queryParam:"style=form,explode=true,name=reportDate"`
}

type GetCompaniesCompanyIDReportsAgedDebtorSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDReportsAgedDebtorRequest struct {
	PathParams  GetCompaniesCompanyIDReportsAgedDebtorPathParams
	QueryParams GetCompaniesCompanyIDReportsAgedDebtorQueryParams
	Security    GetCompaniesCompanyIDReportsAgedDebtorSecurity
}

type GetCompaniesCompanyIDReportsAgedDebtorResponse struct {
	CodatDataContractsDatasetsAgedDebtorOutstandingICollectionReport *shared.CodatDataContractsDatasetsAgedDebtorOutstandingICollectionReport
	ContentType                                                      string
	StatusCode                                                       int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidreportsageddebtor.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdReportsAgedDebtorPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdReportsAgedDebtorQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=numberOfPeriods" })
  numberOfPeriods?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=periodLengthDays" })
  periodLengthDays?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reportDate" })
  reportDate?: Date;
}

export class GetCompaniesCompanyIdReportsAgedDebtorSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdReportsAgedDebtorRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdReportsAgedDebtorPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdReportsAgedDebtorQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdReportsAgedDebtorSecurity;
}

export class GetCompaniesCompanyIdReportsAgedDebtorResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsAgedDebtorOutstandingICollectionReport?: shared.CodatDataContractsDatasetsAgedDebtorOutstandingICollectionReport;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}